### PR TITLE
Fix: Option to delete a kudos message - MEED-1890 - Meeds-io/meeds#781

### PR DESCRIPTION
**Pull request by AzmiTouil**
This PR will allow the space host to delete the kudos activity

### CODE REVIEW

File: kudos-services/src/main/java/org/exoplatform/kudos/listener/KudosActivityListener.java [hunk 1]
```diff
@@ -46,11 +54,31 @@ public void updateActivity(ActivityLifeCycleEvent activityLifeCycleEvent) {
...
   }
 
+  @Override
+  public void deleteActivity(ActivityLifeCycleEvent activityLifeCycleEvent) {
+    ExoSocialActivity activity = activityLifeCycleEvent.getSource();
+    List<Kudos> linkedKudosList = kudosService.getKudosListOfActivity(activity.getId());
+    if (!StringUtils.equals(activity.getType(), KUDOS_ACTIVITY_COMMENT_TYPE) && linkedKudosList.isEmpty()) {
+      return;
+    }
+    deleteLinkedKudos(linkedKudosList);
```
Comment (boubaker) [inline kudos-services/src/main/java/org/exoplatform/kudos/listener/KudosActivityListener.java:64]: What if the activity has sub kudos ? It's better to search for all kudos in the activity and cancel them.

```suggestion
    List<Kudos> linkedKudosList = kudosService.getKudosListOfActivity(activity.getId());
    deleteLinkedKudos(linkedKudosList);
    if (!StringUtils.equals(activity.getType(), KUDOS_ACTIVITY_COMMENT_TYPE) && linkedKudosList.isEmpty()) {
      return;
    }
```

or if you don't mean to cancel all kudos comment, when deleting an activity, then just retrieve the list after the condition test:

```suggestion
    if (!StringUtils.equals(activity.getType(), KUDOS_ACTIVITY_COMMENT_TYPE) && linkedKudosList.isEmpty()) {
      return;
    }
    List<Kudos> linkedKudosList = kudosService.getKudosListOfActivity(activity.getId());
    deleteLinkedKudos(linkedKudosList);
```

Comment (boubaker) [inline kudos-services/src/main/java/org/exoplatform/kudos/listener/KudosActivityListener.java]: Does this the same thing ?

```suggestion
    if (!linkedKudosList.isEmpty()) {
      deleteLinkedKudos(linkedKudosList);
    }
```

Comment (AzmiTouil) [reply kudos-services/src/main/java/org/exoplatform/kudos/listener/KudosActivityListener.java]: yes, right!

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by boubaker — COMMENTED** [review]
Please add Unit Test for newly introduced methods

**Review by boubaker — APPROVED** [review]

**Pull request merged**